src/builtins.c

```diff
@@ -411,13 +415,19 @@ static uintptr_t NOINLINE jl_object_id__cold(jl_datatype_t *dt, jl_value_t *v) J
     }
     if (dt->name->mutabl)
         return inthash((uintptr_t)v);
-    return immut_id_(dt, v, dt->hash);
+    return immut_id_(dt, v, dt->hash, object_table);
 }
 
-JL_DLLEXPORT inline uintptr_t jl_object_id_(jl_value_t *tv, jl_value_t *v) JL_NOTSAFEPOINT
+JL_DLLEXPORT inline uintptr_t jl_object_id_(jl_value_t *tv, jl_value_t *v, htable_t *object_table) JL_NOTSAFEPOINT
 {
+    if(v != NULL && object_table != NULL) {
```

Comment:
@vtjnash as you pointed out, there are other calls to `jl_object_id_`, and so, I made sure that the `object_table` is not null before trying to retrieve the object. This is the case only when it is called from `jl_object_id_temp`. Even then, while running `make`, I get the same error. Is there any other place where I have to change a function call to this function? Because I'm unable to figure out how to do so.

Reply:
The other calls to `jl_object_id_` should be passing in the object_table also. But this function needs to know to ignore it, except when the call came from `jl_object_id_temp`. Or move the code for accessing the table into `jl_object_id_temp`.

Reply:
@vtjnash I believe there is a direct function call to `jl_object_id_` from `ccall.cpp`. There is a variable called `jl_object_id__func` that is a new instance of a `JuliaFunction` struct. I'm guessing that if `NULL` is passed into that function call in place of an object table, then it will work. But I don't know how to pass `NULL` into that function call.